iOSLearningProject/iOSLearningProject/LandingController.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  LandingController.swift
+//  UIOne
+//
+//  Created by Srushti Sarju Engineer on 19/02/25.
+//
+
+import UIKit
+
+class LandingController: UIViewController {
+    @IBAction func signUpClick(_ sender: Any) {
+        print("Sign Up Button Clicked")
+    }
+    @IBAction func logInClick(_ sender: Any) {
+        print("Login Button Clicked")
+    }
+    @IBOutlet weak var btnSignUp: UIButton!
+    @IBOutlet weak var loginButton: UIButton!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        loginButton.layer.borderWidth = CGFloat(1)
+        loginButton.layer.borderColor = CGColor(red: 0.3, green: 0.6, blue: 0.9, alpha: 1.0)
+        btnSignUp.applyGradient()
+    }
+}
+
+extension UIButton {
```

Comment:
This extension should be in a separate file as it is not an extension on this controller

Reply:
And looking at the implementation this seems to be a specific styling, so you can inject that gradient from function and then use it. the intent of extension function is for general purpose applications and not specific usecases